### DISCUSSION TITLE
BRouter increase maximum speed factor

### DISCUSSION THREAD

**Discussion by Leeb64** [devemux86/cruiser#462] [Q&A] [answered]
I often notice that the planned routes make strange, nonsensical detours.
By increasing the speed of the options, these detours become less frequent.
If I set the speed in the BRuoter WebClient to 500km/h, these detours almost disappear completely!
Could this speed setting be expanded accordingly in the BRouter settings in the cruiser? Up to 1000km/h?

![Screenshot 2024-02-29 at 10-41-35 BRouter Web Client 250](https://github.com/devemux86/cruiser/assets/120722053/22ce06a8-8a54-4b1d-9e55-9e425c3dbb37)

![Screenshot 2024-02-29 at 10-45-58 BRouter Web Client 500](https://github.com/devemux86/cruiser/assets/120722053/a3fee392-2f38-45eb-9f74-f84c3f909de5)

**Comment by devemux86** [comment]
@Leeb64 thanks for the suggestion.

This speed should be related to the speed in OpenStreetMap database,
but is not used directly inside the profiles, but inside the BRouter code.

Please provide the BRouter link of the example.

**Reply by Leeb64** [reply]
The problem is that the OSM map data is often incomplete.
You can compensate for this a bit with tricks like this...
There are areas where no speed information is available!
https://brouter.de/essbee/#map=14/44.2283/4.9319/OpenTopoMap&lonlats=4.902477,44.241152;4.943161,44.206338
Cases like this happen often!

https://brouter.de/essbee/#map=13/44.2245/4.3042/OpenTopoMap&lonlats=4.289131,44.234735;4.26424,44.219626
https://brouter.de/essbee/#map=13/44.1755/4.2545/OpenTopoMap&lonlats=4.238319,44.171569;4.243813,44.156621
Change Speed from 200 to 600

**Comment by devemux86** [comment]
@Leeb64 I tested all examples and get the same result with speed 160 / 200 / 600,
using either Cruiser driving profile or BRouter car profiles (with default options).

What profile do you use? Do you also change some other profile options?

**Reply by Leeb64** [reply]
[profile.brf.txt](https://github.com/devemux86/cruiser/files/14446530/profile.brf.txt)

**Reply by devemux86** [reply]
@Leeb64 I can increase the max speed or weight.

However you have enabled most avoid options and the (experimental) advanced options.
It is expected that there will be strange things in the routing. :slightly_smiling_face:

**Reply by Leeb64** [reply]
Motorcycle routes with lots of curves!

Speed ​​setting has the greater impact

**Reply by devemux86** [reply]
@Leeb64 the thing is that with so many options enabled,
at some point they start fighting each other, who will win.

**Reply by Leeb64** [reply]
Achieving the optimal setting is tricky.
But it works!

**Reply by Leeb64** [reply]
I once built a Kurviger offline profile.
To test..
[Extra KURVIG.brf.txt](https://github.com/devemux86/cruiser/files/14469491/Extra.KURVIG.brf.txt)
Set a waypoint approximately every 100km as the crow flies

**Comment by devemux86** [comment]
> Motorcycle routes with lots of curves!

Cruiser will offer another exciting new feature in route planning.

We will be able to discuss more about this in the near future. :slightly_smiling_face:

**Comment by MotoUKRider** [comment]
This is an interesting feature/fix - the knock'on effect is that the wide-ranging scale of the adjustment slider now becomes problematic for users wishing to set 40-50-60-70(mph).  I plotted a simple A-to-B route that defaults to the "big" A-roads and then lowered (from around 100) the slider to 60'ish but found it fiddly.  My route altered, as expected, to put me along the smaller and more fun roads.

(lack of automatic route recalculate upon changing a routing option is still a minor "meh")

Is there any logical reason (not including OSM errors and aircraft profiles!) that we would need to set such high values?

**Reply by devemux86** [reply]
`vmax` and `totalweight` are not used directly inside the profile rules.
They are used in BRouter code, so it's more difficult to see how they are used in calculations.

Note: BRouter website also does not have an upper limit in `vmax` option.

For most users the avoid options in 1st screen should be enough.
Along with `vmax`, although with a different name to be more understandable (?),
as indeed smaller speeds use smaller and more fun roads.

**Reply by devemux86** [reply]
Other websites like [Valhalla](https://valhalla.openstreetmap.de/) have upper limit in max speed (250 km/h).

**Reply by MotoUKRider** [reply]
Yes, the 1st options should be enough.  From my experiences, setting `vmax` to ~65 keeps me off a dual-carriageway (70mph).  It also means that you can reduce the value (but keep the road types) should you want a more leisurely ride, perhaps Primary/Secondary roads but only at 40mph (that would allow you to deliberately ride through towns).

**Reply by Leeb64** [reply]
This speed setting actually has nothing to do with the speed actually driven.
As with so many BRouter settings, it's somehow mysterious...

**Reply by Leeb64** [reply]
In the BRouter Web Client you can set much higher values...???(>10000km/h)
I have no idea what the developers were thinking.
Depending on the criteria set, a high “value” must be set otherwise stupid detours will occur.

**Reply by devemux86** [reply]
For now let's remove the unit and change the name to "Speed factor" to avoid confusion.

**Comment by devemux86** [comment, marked as answer]
BRouter increased maximum speed factor option is available in [Cruiser 3.0.41](https://play.google.com/store/apps/details?id=gr.talent.cruiser).

**Reply by bios4** [reply]
I don't get what this factor/setting is meaning now in practice... 🤔🤨

The "Speed factor" is set to "110" (with unit [km/h]; why is it a "factor" then?) after updating to 3.0.41 on my device.
What does that now "mean" in simple words for routing?

**Reply by devemux86** [reply]
@bios4 max speed (and weight) is used internally in BRouter code.

Lower max speed factor should use roads with smaller max speeds.

**Reply by bios4** [reply]
Ah - so this is somehow a "use only streets with up to max. [value of speed factor] km/h speed limit" setting.

E.g. when the speed limit on highways is 130 km/h (Austria) and 110 km/h is set on my device, the routing will not take highways into account, without having the explicit need to set them as an avoidance?

Something like this?

**Reply by devemux86** [reply]
Something like this. Need to make tests to see how exactly it works.
(better without avoid options, so can see its results alone)

It is often simpler to use only the standard avoid options with specific road types.
